lib/pure/dynlib.nim

```diff
@@ -17,75 +17,69 @@
 ## Loading a simple C function
 ## ---------------------------
 ##
-## The following example demonstrates loading a function called 'greet'
+## The following example demonstrates loading a function called `greet`
 ## from a library that is determined at runtime based upon a language choice.
-## If the library fails to load or the function 'greet' is not found,
+## If the library fails to load or the function `greet` is not found,
 ## it quits with a failure error code.
 ##
-## .. code-block::nim
-##
-##   import std/dynlib
-##
-##   type
-##     greetFunction = proc(): cstring {.gcsafe, stdcall.}
-##
-##   let lang = stdin.readLine()
-##
-##   let lib = case lang
-##   of "french":
-##     loadLib("french.dll")
-##   else:
-##     loadLib("english.dll")
-##
-##   if lib == nil:
-##     echo "Error loading library"
-##     quit(QuitFailure)
-##
-##   let greet = cast[greetFunction](lib.symAddr("greet"))
-##
-##   if greet == nil:
-##     echo "Error loading 'greet' function from library"
-##     quit(QuitFailure)
-##
-##   let greeting = greet()
-##
-##   echo greeting
-##
-##   unloadLib(lib)
-##
+runnableExamples:
```

Comment:
```suggestion
runnableExamples("-r:off"):
```

Reply:
Good idea, didn't think of that.

Though I'm not really happy with the example in general, especially the error handling... What would be more appropriate here?

Reply:
the example is indeed bad, for many reasons. Please improve as you see fit.

I'd like to make it a truly runnableExample without -r:off, it's not too hard but would be very valuable as a self contained working minimal example for library creation + loading.

untested but something similar to this should work:

```nim
import std/[compilesettings, os, strformat, strutils]
const dir = querySetting(SingleValueSetting.nimcacheDir)
const lib = dir / (DynlibFormat % "tdynlib")
let cmd = fmt"{getCurrentCompilerExe()} c --app:lib -o:{lib} {currentSourcePath}"
assert execShellCmd(cmd) == 0, cmd

etc: call loadLib(lib), ...
```

Reply:
That seems far too complex. Examples should be simple, even at the expense of not being tested at run-time.

Reply:
doesn't have to be done in this PR, but IMO such examples are helpful, otherwise you're just shifting the difficulty onto the user for figuring out how to piece together a working example.

One thing we can do though to simplify this example is to instead use lib = `$nim/lib/libnimrtl.$ext` which should be built in CI, so we at least don't have to show the library creation part here.

Reply:
Agreed, if we can abstract away the creation of this dynlib into a `examples.buildGreetNimLib()` then we can avoid overwhelming the user with irrelevant code and give them the time to slowly understand all the pieces. Plus we get another example of how a dll can be created in Nim.

Reply:
ok that's even better than relying on `libnimrtl`, as it's self contained.
so how about split in 2 runnableExamples: the 1st one builds the lib, and the 2nd one loads it.